### PR TITLE
Truncate exception telemetry's parsedStack when there is a deep stack trace exceeding 65536 bytes

### LINKED ISSUE #482: ExceptionTelemetry with large / deep stack traces are silently omitted
When using the Logback appender ApplicationInsightsAppender and so indirectly the components of the applicationinsights-core artifact from
   com.microsoft.azure:applicationinsights-logging-logback:1.0.9
   com.microsoft.azure:applicationinsights-core:1.0.9
where is a problem with log statements that include an exception with deep / long stack traces (ExceptionTelemetry).

Deep stack traces are being serialized as event JSON object correctly which has been verified by debugging the applicationinsights-core code.
But when the resulting JSON of the parse stack (shown in Insights Analytics as "details.parsedStack" and stored in Java class ExceptionTelemetry.ExceptionData.ExceptionDetails.parsedStack) exceeds as certain size (which seems to be 32k according to https://github.com/Microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Docs/ExceptionData.md), the complete log statement is omitted and not shown in Azure Insights Analytics.

I'm not 100% sure if the ExceptionTelemetry is omitted by applicationinsights-core or by Azure Insights instance itself.
But everything indicates that Azure Insights is receiving the JSON and omitting it remotely.

Omitting an ExceptionTelemetry silently is a very problematic behaviour from a development and operations point of view.
The artifact applicationinsights-core or Application Insights itself should know and respect the Azure Insigts limits and provide a robust behaviour:
If stack traces exceed the given limits, at least the log statement without the deep "parsed stack trace" should be stored and shown in Analytics.
Otherwise application exceptions will never be recognized which is a no-go for a monitoring platform.

### PR DESCRIPTION
**Pull request by heyams**
2021-05-20 17:44:13.245-07 DEBUG c.m.applicationinsights.agent - ############# current stack depth: 8943
2021-05-20 17:44:13.266-07 DEBUG c.m.a.telemetry.ExceptionTelemetry - parsedStack is exceeding 65536 bytes capacity. It is truncated from full **1024** frames to partial **308** frames.
@trask method call depth (8943) does not equal to the number of stack frames (total 1024). What i showed you earlier in Fiddler is the full stack.

When we add stack frame to ExceptionTelemetry, it's already in the reversed order.

Fix #482

### CODE REVIEW

**Comment by trask** [comment]
can you test with an exception that is captured by auto-instrumentation? I think that will go through a different code path and not get fixed by this change

**Comment by heyams** [comment]
Breeze accepts exceptions captured by auto-instrumentation that exceed 65536 bytes.  I've sent an email to Oleg to confirm if that's a bug.. because of it's insistent.  [here](https://github.com/cijothomas/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/ExceptionDetails.bond#L25-L27) clearly indicates that both parsedStack and stack should have the same constraint.. but based on my testing, "stack" doesn't have this constraint.  that's why it doesn't have any impact on auto-instrumentation yet.

**Review by trask — APPROVED** [review]

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/ExceptionTelemetry.java [hunk 1]
```diff
@@ -215,4 +225,16 @@ public String getEnvelopName() {
     public String getBaseTypeName() {
         return BASE_TYPE;
     }
+
+    /***
+     * @param stackFrame
+     * @return the stack frame length for only the strings in the stack frame.
+     */
```
Comment (trask) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/ExceptionTelemetry.java:232]:
```suggestion
     */
     // this is the same logic used to limit length on the Breeze side
```

**Review by trask — APPROVED** [review]

**Comment by heyams** [comment]
@trask test:smoke:testApps:WebFlux kept failing.. i reran the job once.. i will let you take a look.. this is keep coming back.

**Pull request merged**